1023/en.md

```diff
@@ -0,0 +1,58 @@
+# Problem Name: Discovering Permutations
+## [LightOj-1023](https://lightoj.com/problem/discovering-permutations)
+## Problem Summary:
+You are given two integer numbers which are N and K.
+Here, N denoted by the first N English capital letters and K denote by the number of permutations.
+If the number of all permutations is less than K then print all of them.
+
+## prerequisite:
+ * [Recurssion](https://www.geeksforgeeks.org/recursion/)
+ * [STL (next_permutation)](https://www.geeksforgeeks.org/stdnext_permutation-prev_permutation-c/)
+
+## Idea and Solution:
+At first, Observe the process on how to work the `next_permutation` method in this code.<br>
+Let, take a string `X=" ABCDEF"` and it's already lexicographically sorted by ascending order.<br>
+The following algorithm generates the next permutation lexicographically after a given permutation.that's means `next_permutation`function return the next immediate larger permutation than the current permutation.
+It changes the given permutation in-place.<br>
+some permutation order:<br>
+ABCDEF<br>
+ABCDFE<br>
+ABCEDF<br>
+ABCEFD<br>
+...<br>
+FEDCBA<br>
+
+**Time Complexity: O(n*n!)** per test_case<br>
```

Comment:
Shouldn't this be O(nk) per test case?

Reply:
Oops, It's a mistake. 
Overall time complexity is O(n) for next_permutation
and per test case time complexity is O(nk).